config/files/usr/share/ublue-os/just/60-custom.just

```diff
@@ -177,3 +177,13 @@ toggle-anticheat-support:
     else 
         echo "The sysctl hardening file is missing the ptrace_scope setting."
     fi
+
+
+# Detect microarchitecture support and apply highest supported flatpak malloc hwcap
+set-flatpak-malloc-hwcap:
+    #!/usr/bin/pkexec /usr/bin/bash
+    uarches=$(/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help | grep '(supported, searched)' | cut -d'v' -f2)
+    bestuarch=${uarches:0:1}
```

Comment:
can you either add a comment explaining this or make it self-documenting?

Reply:
`/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help` prints help info for the ld.so linker. Helpfully, it also reports detected glibc-hwcaps. I grep those, then use cut and the substring selection (:0:1) to get an integer 2-4 corresponding to the system's highest supported hwcap. On an ancient x86_64 system with no v2 support, the bestuarch variable is instead left blank. The if statement checks if it is blank, and if not, it updates the LD_PRELOAD override to the hwcap with our 2-4 integer.